app/src/main/java/com/codepath/chefster/activities/MealLaunchActivity.java

```diff
@@ -31,41 +33,107 @@
+        dishesAdapter = new SmallDishAdapter(this, chosenDishes);
         dishesRecyclerView.setAdapter(dishesAdapter);
         // Setup layout manager for items with orientation
         LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
         // Attach layout manager to the RecyclerView
         dishesRecyclerView.setLayoutManager(layoutManager);
+    }
 
+    private void setupSpinners() {
         // Create an ArrayAdapter using the string array and a default spinner layout
         ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this,
                 R.array.amount_array, android.R.layout.simple_spinner_item);
         // Specify the layout to use when the list of choices appears
         adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
         // Apply the adapter to the spinner
         pansSpinner.setAdapter(adapter);
-        pansSpinner.setSelection(0);
+        pansSpinner.setSelection(numberOfPans - 1);
+        pansSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                numberOfPans = i + 1;
+                calculateCookingTime();
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+
         potsSpinner.setAdapter(adapter);
-        potsSpinner.setSelection(0);
+        potsSpinner.setSelection(numberOfPots - 1);
+        potsSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                numberOfPots = i + 1;
+                calculateCookingTime();
+            }
 
-        onePersonTextView.setBackgroundColor(chosenColor);
-        onePersonTextView.setTextColor(unchosenColor);
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+    }
+
+    /**
+     * This method calculates the time it will take to cook this meal sequentially (without the app)
+     * and how much it will take 1/2 people to cook it with the help of the app
+     */
+    private void calculateCookingTime() {
+        // Prep time is for steps that demand preparation are defined as "Busy" steps, and Cooking time
+        // is for steps that demands only a partial attention, like paste on stove or baking something
+        // in the oven
+        int totalPrepTime = 0, totalCookingTime = 0, totalOptimizedTime = 0, totalAggregatedTime = 0;
+        for (Dish dish : chosenDishes) {
+            totalOptimizedTime += Math.max(dish.getCookingTime(), (dish.getPrepTime() / numberOfPeople));
```

Comment:
Yeah that's a part of the logic. The idea is that a preparation time is a busy time where you have to work and can't do another prep step while doing one already. If you put something in the oven for 60 minutes, and preparation time is 20, the whole thing is still going to take 60 minutes, but you could do the prep steps while its in the oven. Hope that's clear